src/runtime_src/core/common/device.h

```diff
@@ -292,6 +293,8 @@ class device : public ishim
 
   virtual void reset(query::reset_type&) const {}
 
+  virtual void load_xclbin(const MemoryBuffer &) const{}
```

Comment:
This is not commented.  What is this?   xclbin loading is done with xrt::xclbin objects after 4745 pull request.  It looks like this PR is bypassing that.    This function may be fine, but it is confusing that the function seems to be loading xclbins but in a different way than other code.  What does it mean to load an xclbin with this function?   Should the function reflect that loading is via xclmgmt servicing something different?

Should the default behavior be silence?  What does that mean?  Is an exception better?

It looks like std:;vector<char> is not  really used, so it should be const char* and caller should call the function with vector.data().